ethcore/src/engines/authority_round/mod.rs

```diff
@@ -574,8 +577,15 @@ fn verify_timestamp(step: &Step, header_step: u64) -> Result<(), BlockError> {
 			// NOTE This error might be returned only in early stage of verification (Stage 1).
 			// Returning it further won't recover the sync process.
 			trace!(target: "engine", "verify_timestamp: block too early");
-			let oob = oob.map(|n| SystemTime::now() + Duration::from_secs(n));
-			Err(BlockError::TemporarilyInvalid(oob).into())
+
+			let now = SystemTime::now();
+			let found = now.checked_add(Duration::from_secs(oob.found)).ok_or(BlockError::TimestampOverflow)?;
+			let max = oob.max.and_then(|m| now.checked_add(Duration::from_secs(m)));
```

Comment:
I wonder if we should return error on failing `checked_add` here (same for min but max would have failed before). I do not know the authority round logic enough to say.
At the same time if it goes over max value we got a natural bound check from `checked_add` so it is probably not an issue.

Reply:
I think we shouldn't because you will get an error message something like: 

`OutOfBounds { min: None, max: None, found: val }`

which should be more informative than `TimestampOverflow`

However I think it would be better represent those `OutOfBounds` as Duration instead because they are easier to reason about i.e, will always represent seconds as u64!

Thoughts?

Reply:
Maybe `saturating_add` would fit better, my fear was that we were removing a upper bound by making this change but thinking twice it goes directly into error so it does not matter and you are right it only depends upon what we want to display, so it is ok as it is.

Reply:
Right, I get your point saturating_add is a good idea but is very messy for SystemTime